ccmlib/common.py

```diff
@@ -407,6 +402,26 @@ def make_cassandra_env(install_dir, node_path, update_conf=True, hardcode_java_v
     return env
 
 
+_version_regexp = re.compile('[0-9.]+')
+
+def is_jvm_dir_of_version_id(java_version, path):
+    if not path.is_dir():
+        return False
+    if 'jdk' not in path.name:
```

Comment:
this is also fragile, and difficult to comprehend, so instead of parsing the version number encoded in the path, i'd suggest using something like
```py
def get_jvm_spec_version(java: pathlib.Path) -> Optional[str]:
    properties = subprocess.check_output([java.as_posix(), '-XshowSettings:properties', '-version'], stderr=subprocess.STDOUT)
    for line in properties.splitlines():
        parts = line.strip().decode().split('=', 1)
        if len(parts) < 2:
            continue
        name, value = [part.strip() for part in parts]
        if name == 'java.specification.version':
            return value
    return None


known_jvm_names = {
    '8': ['1.8', '8'],
    '11': ['11']
}
for java_path in pathlib.Path('/usr/lib/jvm/').iterdir():
    if not java_path.is_dir():
        continue
    java = java_path / 'bin' / 'java'
    if not java.exists():
        continue
    jvm_version = get_jvm_spec_version(java)
    if jvm_version in known_jvm_names[hardcode_java_version]:
        env['JAVA_HOME'] = str(java_path)
        break
```

also, instead of using "fix" in the title of the commit message, would be better to be more specific. in this case, could be

> ccmlib/common: prevent mistaking JVM 18 as JVM 8
>
> // and explain that we use "in" to check the versions, and "jre-18-openjdk" includes "8-openjdk", so the former is considered as "8-openjdk" by mistake

Reply:
Great idea, done.
Commit message also fixed to be more detailed.